### PR TITLE
Make `@types/express` a peerDependency

### PR DESCRIPTION
**Pull request by daogrady**
Fixes https://github.com/cap-js/cds-types/issues/375

This allows users to freely choose between `@types/express` v4 and v5. This is in line with how we handle things in [`@sap/cds`](https://www.npmjs.com/package/@sap/cds?activeTab=code). I have verified that all uses of the `express` package within our type declarations work in v5 as well.
Starting (again) with npm7, [peerDependencies are automatically](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#peerdependencies) installed if they are not resolvable otherwise, so this should not effect our JS users who have no explicit dependency on `@types/express`.

### CODE REVIEW

**Review by chgeo — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -33,10 +33,8 @@
     "dependencies": "node ./scripts/postinstall.js"
   },
   "peerDependencies": {
-    "@sap/cds": "^8.0.0"
-  },
-  "dependencies": {
-    "@types/express": "^4.17.21"
+    "@sap/cds": "^8.0.0",
+    "@types/express": ">=5"
```
Comment (hakimio) [inline package.json:37]: @daogrady 
Just curious: why did you change `>=4` to `>=5`? Now the changelog entry is not correct too.

Comment (daogrady) [reply package.json:37]: You're right, that's obviously incorrect, thanks for pointing it out!
